internal/controller/postgrescluster/instance_test.go

```diff
@@ -1573,3 +1576,178 @@ func TestFindAvailableInstanceNames(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcileInstanceSetPodDisruptionBudget(t *testing.T) {
+	env, cc, config := setupTestEnv(t, ControllerName)
+	t.Cleanup(func() { teardownTestEnv(t, env) })
+
+	r := &Reconciler{}
+	ctx, cancel := setupManager(t, config, func(mgr manager.Manager) {
+		r = &Reconciler{
+			Client:   cc,
+			Owner:    client.FieldOwner(t.Name()),
+			Recorder: mgr.GetEventRecorderFor(ControllerName),
+			Tracer:   otel.Tracer(t.Name()),
+		}
+	})
+	t.Cleanup(func() { teardownManager(cancel, t) })
+
+	foundPDB := func(
+		t *testing.T,
+		cluster *v1beta1.PostgresCluster,
+		spec *v1beta1.PostgresInstanceSetSpec,
+		minAvailable int32,
+	) bool {
+		pdb, err := r.generatePodDisruptionBudget(
+			cluster,
+			naming.InstanceSetPodDisruptionBudget(cluster, spec),
+			minAvailable,
+			naming.ClusterInstanceSet(cluster.Name, spec.Name),
+		)
+		assert.NilError(t, err)
+		got := &policyv1beta1.PodDisruptionBudget{}
+		err = r.Client.Get(ctx, client.ObjectKeyFromObject(pdb), got)
```

Comment:
⛏️ Similar here; can we omit the call to `generatePodDisruptionBudget`?

```suggestion
		err = r.Client.Get(ctx, naming.AsObjectKey(naming.InstanceSetPodDisruptionBudget(cluster, spec)), got)
```